vstsBuild/vstsBuild.psm1

```diff
@@ -80,13 +82,24 @@ function Invoke-VstsPublishBuildArtifact
             $extension = [System.io.path]::GetExtension($fileName)
             if($extension -ieq '.zip')
             {
-                Expand-Archive -Path $fileName -DestinationPath (Join-Path $destinationPath -ChildPath $leafFileName)
+                $unzipPath = (Join-Path $destinationPath -ChildPath $leafFileName)
+                if($Variable)
+                {
+                    Write-VstsInformation -message "Setting VSTS variable '$Variable' to '$unzipPath'"
+                    Write-Host "##vso[task.setvariable variable=$Variable]$unzipPath"                    
```

Comment:
per our discussion, set the variable as env variable in the current process.
Please also remove the trailing spaces.